website/project/model.py

```diff
@@ -75,16 +74,9 @@ def has_anonymous_link(node, auth):
     :param str link: any view-only link in the current url
     :return bool anonymous: Whether the node is anonymous to the user or not
     """
-    view_only_link = auth.private_key or request.args.get('view_only', '').strip('/')
```

Comment:
For example, at the first call site listed above, we can see *prima facie* that `auth` is constructed from `request.args.get('view_only')`. Therefore, it seems safe to think—modulo the `.strip('/')`—that `auth.private_key` and `request.args.get('view_only', '').strip('/')` are guaranteed to evaluate the same in all cases (given some further safe assumptions, such as that the request is not mutated between the two reads of it).